packages/puppeteer-base/src/sudo.rs

```diff
@@ -94,6 +96,62 @@ where
         Ok(deposits)
     }
 
+    pub fn sudo_delegations_and_balance_kv_query_result(
+        &self,
+        deps: DepsMut<NeutronQuery>,
+        env: Env,
+        query_id: u64,
+        version: &str,
+    ) -> NeutronResult<Response<NeutronMsg>> {
+        let chunks_len = self
+            .delegations_and_balances_query_id_chunk
+            .keys(deps.storage, None, None, Order::Ascending)
+            .collect::<Vec<_>>()
+            .len();
+        let chunk_id = self
+            .delegations_and_balances_query_id_chunk
+            .load(deps.storage, query_id)?;
+        let (remote_height, kv_results) = {
+            let registered_query_result = get_raw_interchain_query_result(deps.as_ref(), query_id)?;
+            (
+                registered_query_result.result.height,
+                registered_query_result.result.kv_results,
+            )
+        };
+        let data: BalancesAndDelegations = PuppeteerReconstruct::reconstruct(&kv_results, version)?;
+        let new_state = match self
+            .delegations_and_balances
+            .may_load(deps.storage, &env.block.height)?
+        {
+            Some(mut state) => {
+                state
+                    .data
+                    .delegations
+                    .delegations
+                    .extend(data.delegations.delegations);
+                // insert chunk_id to collected_chunks and uniq
+                state.collected_chunks.push(chunk_id);
+                state.collected_chunks.sort();
+                state.collected_chunks.dedup();
```

Comment:
I see you dedup collected_chunks and I agree with that. Don't you also have to dedup delegations, or I am being wrong about that?

Reply:
the validators in chunks are separated and merging answers for the chunks is no other than the prev logic with having only one ICQ

Reply:
Well, if a chunk gets pushed twice, then state.data.delegations would have duplicate data, and we don't want that to happen. Maybe instead we should simply do nothing when relayer submits a chunk that we already have?